src/nvim/option.c

```diff
@@ -6987,22 +6985,34 @@ static int opt_strings_flags(
 {
   unsigned int new_flags = 0;
 
-  while (*val) {
+  if (*val == NUL) {
```

Comment:
`if (*val == NUL) { return FAIL; }` should be enough no  ?

Reply:
There seem to be some options with a string forms that allow the empty and some that don't. For example, the `ambw` option allows empty, so `set ambw=` is allowed. On the other hand, `set wak=` will give an `E474` error because `wak` does not allow empty.

Until now, for the do not allow empty option, we had to check if the caller is empty, as shown in the code below. This PR change checks whether the `p_option_values` array contains a `""` or not. Therefore, `if (*val == NUL) { return FAIL; }` is not sufficient.

https://github.com/neovim/neovim/blob/f3d0a1741ef69222ab7893ce4f801d26b6b3fb00/src/nvim/option.c#L2747-L2750